src/models/transactions/offerCancel.ts

```diff
@@ -0,0 +1,21 @@
+import { ValidationError } from "../../common/errors";
+import { BaseTransaction, verifyBaseTransaction } from "./common";
+
+export interface OfferCancel extends BaseTransaction {
+    TransactionType: "OfferCancel";
+    OfferSequence: number;
+}
+
+/**
+ * Verify the form and type of an OfferCancel at runtime.
+ * 
+ * @param tx - An OfferCancel Transaction
+ * @returns - Void.
+ * @throws - When the OfferCancel is Malformed.
+ */
+ export function verifyOfferCancel(tx: OfferCancel): void {
+    verifyBaseTransaction(tx)
+
+    if (tx.OfferSequence !== undefined && typeof tx.OfferSequence !== 'number')
```

Comment:
Should having an undefined offerSequence be an error? (This doc says that it's not an error if the specified offer doesn't exist, but it seems like it might require at least a proposed offer to cancel: https://xrpl.org/offercancel.html)

Reply:
Isn't this type verification unnecessary when we have a defined TypeScript type for it? Wouldn't TS catch this error at compile time?

Reply:
The library can also be used by JavaScript apps, which won't get type safety. These verifications are for them.

Even so, I don't think they are necessary or even particularly helpful. I would be in favor of getting rid of these checks, but it's not a big deal. I'll leave it to @mukulljangid @mvadari @natenichols to decide :)

Reply:
We've talked it over with enough folks that we think that the runtime checks are useful.